### PR TITLE
fix(js-connectors): datetime with optional microseconds

### PR DESCRIPTION
**Pull request by jkomyno**
This PR fixes https://github.com/prisma/team-orm/issues/336.

### CODE REVIEW

File: query-engine/js-connectors/js/smoke-test-js/src/test.ts [hunk 1]
```diff
@@ -20,9 +20,9 @@ export async function smokeTest(db: Connector, prismaSchemaRelativePath: string)
   const test = new SmokeTest(engine, db.flavour)
 
   await test.testFindManyTypeTest()
-  await test.createAutoIncrement()
-  await test.testCreateAndDeleteChildParent()
-  await test.testTransaction()
+  // await test.createAutoIncrement()
+  // await test.testCreateAndDeleteChildParent()
+  // await test.testTransaction()
```
Comment (jkomyno) [inline query-engine/js-connectors/js/smoke-test-js/src/test.ts:25]:
```suggestion
  await test.createAutoIncrement()
  await test.testCreateAndDeleteChildParent()
  await test.testTransaction()
```

File: query-engine/js-connectors/js/smoke-test-js/prisma/postgres-neon/schema.prisma [hunk 2]
```diff
@@ -33,8 +33,8 @@ model type_test {
   date_column_null      DateTime?                   @db.Date
   time_column           DateTime                    @db.Time(0)
   time_column_null      DateTime?                   @db.Time(0)
-  datetime_column       DateTime
-  datetime_column_null  DateTime?
+  datetime_column       DateTime                    @db.Timestamp(3)
+  datetime_column_null  DateTime?                   @db.Timestamp(3)
```
Comment (Jolg42) [inline query-engine/js-connectors/js/smoke-test-js/prisma/postgres-neon/schema.prisma:37]: I'm a bit confused 🤔 `datetime_column` looks a lot like `timestamp_column` now (only the `@db.Timestamp()` value differs, 3 vs 0)

We should also test without `@db.Timestamp(3)` like before, maybe?

Comment (Jolg42) [reply query-engine/js-connectors/js/smoke-test-js/prisma/postgres-neon/schema.prisma:37]: Should this be `@db.DateTime(0)`? 

I found that in https://github.com/prisma/prisma-engines/blob/1f499fcd4a1257371e393196da9a30225513b2a3/query-engine/js-connectors/js/smoke-test-js/prisma/mysql-planetscale/schema.prisma#L44

Comment (Jolg42) [reply query-engine/js-connectors/js/smoke-test-js/prisma/postgres-neon/schema.prisma:37]: I guess I'm also confused that the changes are about `postgres-neon` and there is nothing about `mysql-planetscale` (the bug was found when using PlanetScale adapter)

Comment (jkomyno) [reply query-engine/js-connectors/js/smoke-test-js/prisma/postgres-neon/schema.prisma:37]: `@db.DateTime()` doesn't exist on Postgres, whereas MySQL has both `@db.DateTime()` and `@db.Timestamp()`.
While I haven't committed the MySQL schema.prisma, the change should work for both MySQL and Postgres providers.

File: query-engine/js-connectors/js/smoke-test-js/src/test.ts [hunk 3]
```diff
@@ -48,6 +47,61 @@ export async function smokeTest(db: Connector, prismaSchemaRelativePath: string)
...
 
+  async testTypeTest2() {
+    const create = await this.engine.query(`
+      {
+        "action": "createOne",
+        "modelName": "type_test_2",
+        "query": {
+          "arguments": {
+            "data": {}
+          },
```
Comment (jkomyno) [inline query-engine/js-connectors/js/smoke-test-js/src/test.ts:58]: Notice that we're relying on the defaults here.

**Review by Jolg42 — APPROVED** [review]
I'll merge this now and try it tomorrow morning

**Pull request merged**